src/Rings/orderings.jl

```diff
@@ -174,6 +174,7 @@ mutable struct MonomialOrdering{S}
   o::AbsGenOrdering
   is_total::Bool
   is_total_is_known::Bool
+  canonical_matrix::ZZMatrix
```

Comment:
```suggestion
  canonical_matrix::ZZMatrix
  function MonomialOrdering(R::S, o::AbsGenOrdering) where S
     return new{S}(R, o, false, false)
  end
```
or for more flexibility:
```suggestion
  canonical_matrix::ZZMatrix
  function MonomialOrdering(R::S, o::AbsGenOrdering) where S
     return new{S}(R, o, false, false)
  end
  function MonomialOrdering(R::S, o::AbsGenOrdering, is_total::Bool) where S
     return new{S}(R, o, is_total, true)
  end
```
and then adjust the outer constructor below